test/inductor/test_pattern_matcher.py

```diff
@@ -328,28 +328,38 @@ def fn(a, b, c):
                 torch.randn(16, 16, device="cuda"),
                 torch.randn(16, 16, device="cuda"),
                 torch.randn(16, 16, device="cuda"),
+                True,
             ),
             (
                 torch.randn(16, 16, device="cuda"),
                 torch.randn(1, 16, device="cuda"),
                 torch.randn(16, 16, device="cuda"),
+                False,
             ),
             (
                 torch.randn(1, 16, 16, device="cuda"),
                 torch.randn(16, 16, device="cuda"),
                 torch.randn(16, 16, device="cuda"),
+                False,
+            ),
+            (
+                4,
+                torch.randn(16, 16, device="cuda"),
+                torch.randn(16, 16, device="cuda"),
+                False,
             ),
-            (4, torch.randn(16, 16, device="cuda"), torch.randn(16, 16, device="cuda")),
         ]
-        for args in args_list:
+        for a, b, c, should_fuse in args_list:
             torch._dynamo.reset()
             counters.clear()
+            args = (a, b, c)
             e1, e2 = fn(*args)
             a1, a2 = torch.compile(fn)(*args)
             torch.testing.assert_close(a1, e1)
             torch.testing.assert_close(a2, e2)
-            self.assertEqual(counters["inductor"]["pattern_matcher_count"], 2)
-            self.assertEqual(counters["inductor"]["pattern_matcher_nodes"], 4)
+            count, nodes = (2, 4) if should_fuse else (0, 0)
```

Comment:
Note that these cases weren't actually fused previously, it's just that the pattern replaced them with a lowering that did `add + mm`.